rules/regex-shorthand.js

```diff
@@ -1,30 +1,21 @@
 'use strict';
+
 const cleanRegexp = require('clean-regexp');
+
+const {parse, generate, optimize} = require('regexp-tree');
+
 const getDocsUrl = require('./utils/get-docs-url');
 
 const message = 'Use regex shorthands to improve readability.';
 
 const create = context => {
-	return {
-		'Literal[regex]': node => {
-			const oldPattern = node.regex.pattern;
-			const {flags} = node.regex;
-
-			const newPattern = cleanRegexp(oldPattern, flags);
-
-			// Handle regex literal inside RegExp constructor in the other handler
-			if (node.parent.type === 'NewExpression' && node.parent.callee.name === 'RegExp') {
-				return;
-			}
+	const sourceCode = context.getSourceCode();
+	const options = context.options[0] || [
```

Comment:
Use the JSON schema default key for the default values: https://github.com/sindresorhus/eslint-plugin-unicorn/issues/247